### ISSUE TITLE
How Can I Send Message On Specific Skype User

### ISSUE THREAD

**Issue by inornos** [Microsoft/BotBuilder#1670]
I Want Send Message To All My Bot User To Inform Some Detail, I Have All Skype ID Like
json567
fss80support
rajuDiam85
This Is the skype id and i want send message to them all 
so please help me to send message from bot to my all skype id
please share some code
Thanks

**Comment by leroibeukes** [comment]
You can't ... at least not in the way you propose.

Your user's must have had a conversation with you and you must have persisted the information  somewhere. 

Then you can use that to send them a message , eg. 

Let's say you did manage to save the data (as a JSON object) somewhere ... 

JSON data object (pseudo code):
 {
  "conversationUrl": "https://facebook.botframework.com",
  "conversationId": "xxxxx-xxxxxxx",
  "channelId": "facebook",
  "locale": "en-US",
  "botUserId": "xxxxxx",
  "botUserName": "xxxxxx"
}

Then you could use this data to 'resume' the conversation ... 
Disclaimer: There is an alternative method as well (Google 'Resumption Cookie').

//C# code
JObject jData = JObject.Parse(data);
                        var connector = new ConnectorClient(new Uri(jData["conversationUrl"].ToString()));
                        var conversation = new ConversationAccount(true, jData["conversationId"].ToString());
                        var botAccount = new ChannelAccount(jData["botUserId"].ToString(), jData["botUserName"].ToString());
                        IMessageActivity message = Activity.CreateMessageActivity();
                        message.From = botAccount;
                        message.Conversation = conversation;
                        message.ChannelId = jData["channelId"].ToString();
                        message.Text = "This is me Spamming you";
                        message.Locale = jData["locale"].ToString();
                        await connector.Conversations.SendToConversationAsync((Activity)message);

**Comment by inornos** [comment]
hello @leroibeukes
Basically i want create chat support system for our skype end users, Our skype user can get support from our website using **web chat** When **Web Chat** I frame open in website for send message user can directly contact to out support person to get help and also we want to manage conversion in skype software with different user so we can easily find past history to view specific users

**Comment by chrimc62** [comment]
How would you know the user's skype ID from your webchat?

**Comment by leroibeukes** [comment]
The Webchat channel and Skype Channel are two different channels - you would have to have the user identify themselves on both channels for you to be able to 'pick-up where you left-off' ...

Thus when a user starts a conversation in the webchat - you could ask them to 'login/identify' via some or other service/api or similar - you could then use the same mechanism in your Skype channel and that way 'tie/link' the two together. So to give a very rudimentary example :

1. User starts chatting in Webchat.
2. Webchat asks user if it is there first time
3. if not - then agent asks them to authenticate  either in -channel or break out to site and 
    then go to step 4.
4. If so - then agent asks for their 'token' or unique userId (assigned by you)
5. User provides 'token' or unique id and it is assigned to the conversation and state is retreived etc.

6. Same User then uses the Skype channel
7. On Skype User is asked for their 'token' or unique id
8. If User provides it then the state is retrieved and the webchat is associated with the user.
9. If not - then break out etc. (Same as Step3).

So in a nutshell you have to be able to identify the user when he starts engaging with you (regardless of channel) , that way you can retrieve state for the user and history etc.

Hope this helps shape your thinking.

Good luck
Cheers

**Comment by chrimc62** [comment]
If this answer doesn't work for you, please let us know.

**Comment by pranotiB** [comment]
Hi
I have created chat bot using api.ai and I have integrated with skype.
And also im using web hook to check the user data and reply to it accordingly.

But im not able to send message from chat bot to the user. I tried sending conversation object but its not working..
Does anyone know how to do this??

Thank you in advance.

**Comment by EricDahlvang** [comment]
Hi pranotiB,

Thank you for your submission. 

The Microsoft Bot Framework team prefers that **how to** questions be submitted on [Stack Overflow](http://stackoverflow.com/search?q=%23botframework). The official [Bot Framework Github](https://github.com/Microsoft/BotBuilder/issues) repo  is the preferred platform for submitting **bug fixes and feature requests**.

**Comment by pranotiB** [comment]
Okay. Thanks EricDahlvang.

I have posted my query on stack overflow..

**Comment by EricDahlvang** [comment]
Hi @jagufru,

The Skype user needs to first communicate with the bot.   The bot then cashes the address, and can proactively message the user.  This doc explains more: https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-proactive-messages

**Comment by sonisingh9877nov** [comment]
A verbal exchange is a sequence of messages sent among your bot and one or more customers. There are three forms of conversations (also known as scopes) in teams: ... private Conversations among bots and a single consumer. groupchat Chat between a bot and two or greater customers.

https://www.techlinc.in/best-whatsapp-status-quotes/

https://www.techlinc.in/whatsapp-dp-profile-picture/
